### PR TITLE
further fix on cross-cancer expression tab

### PR DESCRIPTION
**Pull request by yichaoS**
1) if there is no RNA Seq V2 and there is RNA Seq, maybe switch to RNA Seq directly.
http://bit.ly/2rFx6eP
2) if no studies were selected by default, display the study selection box.
http://bit.ly/2qBJSqK

### CODE REVIEW

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js [hunk 1]
```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
...
-                                _.each(_d[0].concat(_d[1], _d[2]), function(_obj) {
-                                    if ( _obj.sequenced) {
-                                        if (_obj.mutation_type === "non" ) {
-                                            result_str += _obj.sample_id + "\t" + _obj.study_name + "\t" + "RNA Seq V2" + "\t" + gene[0] + "\t" + "Not Mutated" + "\t" + _obj.profile_data + "\n";
+                    $.when(ccPlots.util.getGeneticProfiles_(_.pluck(_.pluck(window.studies.models, "attributes"), "studyId"))).then(function(_profiles) {
+                        
+                        // if no v2 profile (data) available, set profile selection to v1
+                        if($("#cc_plots_profile_list option").length === 0) {
+                            var _hasV2 = false;
+                            _.each(_.pluck(_profiles, "id"), function(_pid) {
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: You can consider `_.some`

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js [hunk 2]
```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
...
+                                if (_pid.indexOf("rna_seq_v2_mrna") !== -1) _hasV2 = true;
+                            });
+                            if (!_hasV2) $("#cc_plots_profile_list").append("<option value='v1'>RNA Seq</option><option value='v2'>RNA Seq V2</option>");
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: If all studies dont have v2, why do we give v2 option?

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js [hunk 3]
```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
...
+                            else $("#cc_plots_profile_list").append("<option value='v2'>RNA Seq V2</option><option value='v1'>RNA Seq</option>");                            
+                        }
+                            
+                        // default menu settings
+                        document.getElementById("cc_plots_gene_list").disabled = false;
+                        gene = [];
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: Can be removed.

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js [hunk 4]
```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
...
+                                cbio.download.initDownload(final_pdf_str, downloadOptions);
+
+                            });
+                            $("#cc_plots_data_download").click(function() {
+                                var get_tab_delimited_data = function() {
+                                    var result_str = "Sample Id" + "\t" + "Cancer Study" + "\t" + "Profile Name" + "\t" + "Gene" + "\t" + "Mutation" + "\t" + "Value" + "\n";
+                                    _.each(_d[0].concat(_d[1], _d[2]), function(_obj) {
+                                        if ( _obj.sequenced) {
+                                            if (_obj.mutation_type === "non" ) {
+                                                result_str += _obj.sample_id + "\t" + _obj.study_name + "\t" + "RNA Seq V2" + "\t" + gene[0] + "\t" + "Not Mutated" + "\t" + _obj.profile_data + "\n";
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: This text needs to be updated with different RNA Seq versions?

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js [hunk 5]
```diff
@@ -515,80 +520,97 @@ var ccPlots = (function (Plotly, _, $) {
...
+
+                                var downloadOptions = {
+                                    filename: "cross-cancer-plots.pdf",
+                                    contentType: "application/pdf",
+                                    servletName: "svgtopdf.do"
+                                };
+                                cbio.download.initDownload(final_pdf_str, downloadOptions);
+
+                            });
+                            $("#cc_plots_data_download").click(function() {
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: All similar jquery events need to be `unbind` first. After switching RNA version several times, if I click to download DATA, I got multiple files.

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js (hunk 2 above)
Comment (yichaoS) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: It's better to show a message saying we don't have data for that profile that just removing the option quietly.

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js (hunk 3 above)
Comment (yichaoS) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: will do

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js (hunk 4 above)
Comment (yichaoS) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: will fix.

File: portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js (hunk 5 above)
Comment (yichaoS) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: will fix

Comment (yichaoS) [inline portal/src/main/webapp/js/src/cross-cancer-plotly-plots.js]: Added unbind at line 546

**Pull request merged**